### ISSUE TITLE
Gate.io fetch_ohlcv no longer working correctly

### ISSUE THREAD

**Issue by xmatthias** [ccxt/ccxt#11162]
- OS:  linux
- Programming Language version: python3.7+
- CCXT version:  1.66.36

``` python
import ccxt
exchange = ccxt.gateio()
pair = "ETH/USDT"
timeframe = '1h'
since_int = 1641314340000  # 3 days ago ... should obviously be recalculated eventually ... 
candles = exchange.fetch_ohlcv(
    pair, 
    timeframe, 
    since=since_int, 
    limit=1000,
    # params={
    #     'from': 1612134000,
    #     'to': 1612194000,
    # }
    )
```

```
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/ccxt/base/exchange.py in fetch(self, url, method, headers, body)
    657             self.logger.debug("%s %s, Response: %s %s %s", method, url, http_status_code, headers, http_response)
--> 658             response.raise_for_status()
    659 

~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/requests/models.py in raise_for_status(self)
    952         if http_error_msg:
--> 953             raise HTTPError(http_error_msg, response=self)
    954 

HTTPError: 400 Client Error: Bad Request for url: https://api.gateio.ws/api/v4/spot/candlesticks?currency_pair=ETH_USDT&interval=1h&limit=1000&from=1641314340&to=1644914339

During handling of the above exception, another exception occurred:

BadRequest                                Traceback (most recent call last)
/tmp/ipykernel_3067083/764404986.py in <module>
----> 1 candles = exchange.fetch_ohlcv(
      2     pair,
      3     timeframe,
      4     since=since_int,
      5     limit=1000,

~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/ccxt/gateio.py in fetch_ohlcv(self, symbol, timeframe, since, limit, params)
   1647             request['from'] = int(since / 1000)
   1648             request['to'] = self.sum(request['from'], limit * duration - 1)
-> 1649         response = getattr(self, method)(self.extend(request, params))
   1650         return self.parse_ohlcvs(response, market, timeframe, since, limit)
   1651 

~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/ccxt/base/exchange.py in inner(_self, params, context)
    498                 if context is not None:
    499                     inner_kwargs['context'] = params
--> 500                 return entry(_self, **inner_kwargs)
    501             return inner
    502         to_bind = partialer()

~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/ccxt/base/exchange.py in request(self, path, api, method, params, headers, body, config, context)
    549     def request(self, path, api='public', method='GET', params={}, headers=None, body=None, config={}, context={}):
    550         """Exchange.request is the entry point for all generated methods"""
--> 551         return self.fetch2(path, api, method, params, headers, body, config, context)
    552 
    553     @staticmethod

~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/ccxt/base/exchange.py in fetch2(self, path, api, method, params, headers, body, config, context)
    545         self.lastRestRequestTimestamp = self.milliseconds()
    546         request = self.sign(path, api, method, params, headers, body)
--> 547         return self.fetch(request['url'], request['method'], request['headers'], request['body'])
    548 
    549     def request(self, path, api='public', method='GET', params={}, headers=None, body=None, config={}, context={}):

~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/ccxt/base/exchange.py in fetch(self, url, method, headers, body)
    672         except HTTPError as e:
    673             details = ' '.join([self.id, method, url])
--> 674             self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
    675             self.handle_http_status_code(http_status_code, http_status_text, url, method, http_response)
    676             raise ExchangeError(details) from e

~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/ccxt/gateio.py in handle_errors(self, code, reason, url, method, headers, body, response, requestHeaders, requestBody)
   2951         if label is not None:
   2952             feedback = self.id + ' ' + body
-> 2953             self.throw_exactly_matched_exception(self.exceptions['exact'], label, feedback)
   2954             raise ExchangeError(feedback)

~/.pyenv/versions/3.9.7/envs/trade_397/lib/python3.9/site-packages/ccxt/base/exchange.py in throw_exactly_matched_exception(self, exact, string, message)
    563     def throw_exactly_matched_exception(self, exact, string, message):
    564         if string in exact:
--> 565             raise exact[string](message)
    566 
    567     def throw_broadly_matched_exception(self, broad, string, message):

BadRequest: gateio {"label":"INVALID_PARAM_VALUE","message":"Candlestick range too broad. Maximum 1000 data points are allowed per request"}
```

The same exact code works fine on ccxt 1.61.92 (that's what i tested - but it's not necessarily the exact version that broke it.
I suspect this has been introduced in #11019 - as that's the only changes to this method.

**Comment by kroitor** [comment]
Thanks for reporting! Will investigate and fix shortly.

**Comment by kroitor** [comment]
Fixed in CCXT 1.66.39+, turns out, gate.io does not like it when the to-timestamp is in the future. The new build will arrive in ~20 minutes. Thanks for reporting, as always!
